src/tests/system/tests/test_gdm.py

```diff
@@ -0,0 +1,204 @@
+"""
+SSSD Passwordless GDM Tests
+
+:requirement: Passwordless GDM
+"""
+
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.ipa import IPA
+from sssd_test_framework.roles.keycloak import Keycloak
+from sssd_test_framework.topology import KnownTopology
+from sssd_test_framework.utils.authentication import KerberosAuthenticationUtils as kerberos
+
+
+@pytest.mark.topology(KnownTopology.GDM)
+@pytest.mark.builtwith(client="gdm")
+def test_gdm__xidp_login(client: Client, ipa: IPA, keycloak: Keycloak):
+    """
+    :title: Login via GDM with external IdP user
+    :setup:
+        1. Configure IPA for External IdP support
+        2. Add user to IPA as IdP user
+        3. Configure SSSD pam_json_services = gdm-switchable-auth
+    :steps:
+        1. Select user from list (if not listed, use Other and enter user)
+        2. Follow IdP redirect url and follow steps to authorize request
+        3. Complete login in GDM (select Done)
+        4. Confirm user logged in by checking for home screen
+    :expectedresults:
+        1. User selected and login method (EIdP) chosen
+        2. User connects to remote URL with browser and authorized request successfully
+        3. User is logged into system
+        4. User can see login home screen after login
```

Comment:
Can you review these? The steps don't match the procedure you are following in the test

Reply:
Same applies to other test cases

Reply:
I'd minimize it a bit
```
:steps:
    1. Login through GDM by authenticating with Keycloak from an external browser.
:expectedresults:
    1. Login succeeds and user sees the home screen.
```